client-proxy/transports/transport.go

```diff
@@ -0,0 +1,19 @@
+package transports
```

Comment:
We define how transports are configured as an underlying [client](https://github.com/OpenLineage/OpenLineage/tree/main/client) implementation. For example, in Java you can configure a [`Transport`](enLineage/tree/main/client/java#transports) via a [`openlineage.yml`](https://github.com/OpenLineage/OpenLineage/tree/main/client/java#configuration). This can be done as a follow up PR, but I would suggest we move the `client` code under `client/go` that would be used by the proxy; though the proxy is handling strings, the client can allow for `string` type events or (more preferably) define a `OlEvent` object with only top-level properties that are required.

Reply:
The transport is not for the client implementation but the logic for transport. It might use go client or simple http client, but the responsibility of this is not implementing the client, but is to use client in order to send events to the configured backend.

For the second point, I don't think defining an `OLEvent` is preferred. For the client library we should do, because we construct the openlineage event from the integration. We construct the event with information available, and the defined model is helpful. For the proxy, we only pass the event as is but we don't touch it. Having a defined model means that we have to deserialize and serialize again with the risk of missing some information. I would prefer not to have a model for this.

Reply:
> The transport is not for the client implementation but the logic for transport. It might use go client or simple http client, but the responsibility of this is not implementing the client, but is to use client in order to send events to the configured backend.

I don't think the `go` client needs to be its own package, at least not initially, but it would be great to have the `proxy` make use of a `go` client in the log term. For example, the `http` transports is simple enough, but the `proxy` expands to support `kafka` transports as a `backend`, it move the embedded `go` into its own package.

Reply:
I think that this is something we fundamentally have a different idea. We might need separate go client outside of this project if we implement some go integrations, but even with that I don't think it's a good idea to use the client for this use case. Using a defined model brings no benefit but only the friction.

Besides, it's definitely out of scope but I'm not even sure that it's a good idea to make client package internally deal with different transport type. At the moment, the spec only defines the HTTP endpoint and both python and java client implement just that. Do we even plan doing so for java and python clients? I don't think we have a consensus about it.

Reply:
> At the moment, the spec only defines the HTTP endpoint and both python and java client implement just that. Do we even plan doing so for java and python clients? I don't think we have a consensus about it.

Both the both python and java client implement `Kafka` as a transport.